background.js

```diff
@@ -111,6 +127,10 @@ ipcMain.on('k8s-restart', () => {
       .then(() => {
         tray.k8sRestarting();
       })
+      .then(() => {
+        // The desired Kubernetes version might have changed
+        k8smanager = K8s.factory(cfg.kubernetes);
+      })
       .then(() => { k8smanager.start() })
```

Comment:
The state issue on changing k8s versions is due to this line. It needs a return statement before k8smanager.start(). I'll do that in a follow-up PR.